src/Entities/Entity.cpp

```diff
@@ -1878,72 +1878,69 @@ void cEntity::TeleportToCoords(double a_PosX, double a_PosY, double a_PosZ)
 void cEntity::BroadcastMovementUpdate(const cClientHandle * a_Exclude)
 {
 	// Process packet sending every two ticks
-	if (GetWorld()->GetWorldAge() % 2 == 0)
+	if (GetWorld()->GetWorldAge() % 2 != 0)
 	{
-		double SpeedSqr = GetSpeed().SqrLength();
-		if (SpeedSqr == 0.0)
-		{
-			// Speed is zero, send this to clients once only as well as an absolute position
-			if (!m_bHasSentNoSpeed)
-			{
-				m_World->BroadcastEntityVelocity(*this, a_Exclude);
-				m_World->BroadcastTeleportEntity(*this, a_Exclude);
-				m_bHasSentNoSpeed = true;
-			}
-		}
-		else
+		return;
+	}
+
+	if (GetSpeed().HasNonZeroLength())
+	{
+		// Movin'
+		m_World->BroadcastEntityVelocity(*this, a_Exclude);
+		m_bHasSentNoSpeed = false;
+	}
+	else
+	{
+		// Speed is zero, send this to clients once only as well as an absolute position
+		if (!m_bHasSentNoSpeed)
 		{
-			// Movin'
 			m_World->BroadcastEntityVelocity(*this, a_Exclude);
-			m_bHasSentNoSpeed = false;
+			m_World->BroadcastTeleportEntity(*this, a_Exclude);
+			m_LastSentPosition = GetPosition();
```

Comment:
This is the main change, previously `m_LastSentPosition` becomes invalid whenever an entity stops moving for even a single tick. That could cause big desyncs if the movement is very start-stop.